drivers/i2c/i2c_dw.c

```diff
@@ -63,10 +65,14 @@ static inline void i2c_dw_data_ask(const struct device *dev)
 	}
 
 	/* Get the FIFO depth that could be from 2 to 256 from HW spec */
+#ifdef CONFIG_I2C_DW_ARC_AUX_REGS
+	rx_buffer_depth = CONFIG_I2C_DW_FIFO_DEPTH;
+	tx_buffer_depth = CONFIG_I2C_DW_FIFO_DEPTH;
+#else
 	ic_comp_param_1.raw = read_comp_param_1(reg_base);
 	rx_buffer_depth = ic_comp_param_1.bits.rx_buffer_depth + 1;
 	tx_buffer_depth = ic_comp_param_1.bits.tx_buffer_depth + 1;
-
+#endif
```

Comment:
I'm wondering why do we use different approach for DFSS peripheral?

Do we have this `comp_param_1` reg missing in case of DFSS?

Reply:
Yes, DFSS don't have `comp_param_1` register to restore FIFO like property information. We need to manually define that.